cpu/cc2538/include/periph_cpu.h

```diff
@@ -83,6 +83,31 @@ typedef struct {
     uint_fast8_t cfg;       /**< timer config word */
 } timer_conf_t;
 
+/**
+ * @brief   Override resolution options
+ */
+#define HAVE_ADC_RES_T
+typedef enum {
+    ADC_RES_6BIT  = 0xa00,          /**< not supported by hardware */
+    ADC_RES_7BIT  = (0 << 4),       /**< ADC resolution: 7 bit */
+    ADC_RES_8BIT  = 0xb00,          /**< not supported by hardware */
+    ADC_RES_9BIT  = (1 << 4),       /**< ADC resolution: 9 bit */
+    ADC_RES_10BIT = (2 << 4),       /**< ADC resolution: 10 bit */
+    ADC_RES_12BIT = (3 << 4),       /**< ADC resolution: 12 bit */
+    ADC_RES_14BIT = 0xc00,          /**< not supported by hardware */
+    ADC_RES_16BIT = 0xd00,          /**< not supported by hardware */
+} adc_res_t;
+
+/**
+ * @brief Masks for getting data
+ * @{
+ */
+#define SOCADC_7_BIT_RSHIFT        9 /**< Mask for getting data( 7 bits ENOB) */
+#define SOCADC_9_BIT_RSHIFT        7 /**< Mask for getting data( 9 bits ENOB) */
+#define SOCADC_10_BIT_RSHIFT       6 /**< Mask for getting data(10 bits ENOB) */
+#define SOCADC_12_BIT_RSHIFT       4 /**< Mask for getting data(12 bits ENOB) */
```

Comment:
Since there is a RSHIFT required after a conversion result, I suggest encoding the RSHIFT values into the ADC_RES_XXX values. For example:

ADC_RES_7BIT = (0 << 4) && (SOCADC_7_BIT_RSHIFT << 8);

After reading the conversion result, you can:

result = result >> (res >> 8);